subprojects/core/src/main/java/org/gradle/api/file/FileTreeElement.java

```diff
@@ -103,4 +103,9 @@
      * @return The Unix file permissions.
      */
     int getMode();
+
+    /**
+     * @return true if the file is a regular file or directory on disk. False for files in zip or tar files.
+     */
+    boolean isPhysicalFile();
```

Comment:
I don't think this is the right place to do this. We should fix it so we can revisit zipTree/tarTree and make this work without this kind of flag.